orchestra/communication/utils.py

```diff
@@ -0,0 +1,61 @@
+from orchestra.models import CommunicationPreference
+from orchestra.models import StaffBotRequest
+from orchestra.models import StaffingRequestInquiry
+from orchestra.models import StaffingResponse
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def mark_worker_as_winner(worker, task, required_role_counter,
+                          staffing_request_inquiry):
+    staffbot_request = StaffBotRequest.objects.filter(
+        task=task, required_role_counter=required_role_counter)
+
+    # Check whether staffbot request was sent out for this task
+    if not staffbot_request.exists():
+        return
+
+    staffbot_request = staffbot_request.first()
+    if staffing_request_inquiry:
+        staffbot_request = staffing_request_inquiry.request
+
+    staffbot_request.status = StaffBotRequest.Status.COMPLETE.value
+    staffbot_request.save()
+
+    # Mark everyone else as non-winner
+    StaffingResponse.objects.filter(
+        request_inquiry__request=staffbot_request).update(is_winner=False)
+
+    # If staffing request inquiry provided
+    if staffing_request_inquiry:
+        staffing_response = staffing_request_inquiry.responses.all()
+    else:
+        staffing_response = StaffingResponse.objects.filter(
+            request_inquiry__request=staffbot_request,
+            request_inquiry__communication_preference__worker=worker)
+
+    if staffing_response.exists():
+        staffing_response = staffing_response.first()
+        staffing_response.is_winner = True
+        staffing_response.save()
+    else:
```

Comment:
should we remove this else and the code in it? i'm thinking through the reason this case pops up:
- if the Worker responded to an inquiry, then they have a staffing_response and are marked as the winner above
- if the worker was assigned from projman or some other means, then there might not be an inquiry for that worker at all. so they didn't "win" and there was no "inquiry."

can we think of why this code path would happen?

Reply:
i guess the use case is if StaffBot ran, but someone assigned the task to a Worker before they received a request.

i could go either way. if you keep this code around, leave a comment explaining the stuff above :)